### PR TITLE
[SPARK-13070][SQL] Better error message when Parquet schema merging fails

### PR DESCRIPTION
**Pull request by viirya**
Make sure we throw better error messages when Parquet schema merging fails.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala [hunk 1]
```diff
@@ -800,9 +800,21 @@ private[sql] object ParquetRelation extends Logging {
...
+              } catch { case cause: SparkException =>
+                throw new SparkException(
+                  s"Failed merging schema of file ${footer.getFile}:\n${schema.treeString}", cause)
+              }
+            }
+            Iterator.single(mergedSchema)
+          }
         }.collect()
 
     partiallyMergedSchemas.reduceLeftOption(_ merge _)
```
Comment (holdenk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala]: Might we also run into the same merge exception here?

Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala]: yes. I think so. Update this now.

**Comment by viirya** [comment]
retest this please.

**Comment by rxin** [comment]
cc @liancheng

**Comment by liancheng** [comment]
LGTM, thanks for helping fixing this! Will get it merged once I get access to my laptop.

**Comment by rxin** [comment]
I've merged this in master. Thanks.

**Pull request closed without merging**